cognite/neat/_graph/transformers/_classic_cdf.py

```diff
@@ -75,7 +77,31 @@ def get_depth(
             return None
 
 
-class AssetTimeSeriesConnector(BaseTransformer):
+class ClassicConnector(BaseTransformer, ABC):
+    _asset_type: URIRef = DEFAULT_NAMESPACE.Asset
+    _item_type: URIRef
+    _default_attribute: URIRef
+    _connection_type: URIRef
+
+    _select_item_ids = "SELECT DISTINCT ?item_id WHERE {{?item_id a <{item_type}>}}"
+    _select_connected_assets: str = textwrap.dedent("""SELECT ?asset_id WHERE {{
+                              <{item_id}> <{attribute}> ?asset_id .
+                              ?asset_id a <{asset_type}>}}""")
+
+    def __init__(self, attribute: URIRef | None = None) -> None:
+        self._attribute = attribute or self._default_attribute
+
+    def transform(self, graph: Graph) -> None:
+        for item_id, *_ in graph.query(self._select_item_ids.format(item_type=self._item_type)):  # type: ignore[misc]
+            for asset_id, *_ in graph.query(  # type: ignore[misc]
+                self._select_connected_assets.format(
+                    item_id=item_id, attribute=self._attribute, asset_type=self._asset_type
+                )
+            ):
+                graph.add((asset_id, self._connection_type, item_id))
```

Comment:
use `add_triples_in_batch` to batch write triples instead individually writing them